test/e2e/storage/drivers/csi.go

```diff
@@ -573,11 +573,38 @@ func (g *gcePDCSIDriver) GetSnapshotClass(config *storageframework.PerTestConfig
 }
 
 func (g *gcePDCSIDriver) PrepareTest(f *framework.Framework) (*storageframework.PerTestConfig, func()) {
+	testns := f.Namespace.Name
+
+	if framework.ProviderIs("gke") {
+		framework.Logf("The csi gce-pd driver is automatically installed in GKE. Skipping driver installation.")
+
+		// Cleanup CSI driver and namespaces. This function needs to be idempotent and can be
+		// concurrently called from defer (or AfterEach) and AfterSuite action hooks.
+		cleanupFunc := func() {
+			ginkgo.By(fmt.Sprintf("deleting the test namespace: %s", testns))
+			// Delete the primary namespace but it's okay to fail here because this namespace will
+			// also be deleted by framework.Aftereach hook
+			tryFunc(func() { f.DeleteNamespace(testns) })
+
+			// cleanup function has already ran and hence we don't need to run it again.
+			// We do this as very last action because in-case defer(or AfterEach) races
+			// with AfterSuite and test routine gets killed then this block still
+			// runs in AfterSuite
+			framework.RemoveCleanupAction(g.cleanupHandle)
+		}
+		g.cleanupHandle = framework.AddCleanupAction(cleanupFunc)
+
+		return &storageframework.PerTestConfig{
+			Driver:    g,
+			Prefix:    "gcepd",
```

Comment:
random question, why are we using gcepd here? That seems to add confusion between the pd.csi.storage.gke.io and gce-pd provisioners.

I think this is just a test prefix (?) but still...

Reply:
There is an unresolved bug in the e2e test suite that generates binary flags based on driver name. And if you have 2 driver instances with the same name (ie run pd csi externally, the binary gets built with two definitions of the same driver), then it conflicts. https://github.com/kubernetes/kubernetes/issues/90868